internal/service/identitystore/group.go

```diff
@@ -0,0 +1,242 @@
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return create.DiagError(names.IdentityStore, create.ErrActionReading, ResNameGroup, d.Id(), err)
+	}
+
+	d.Set("group_id", out.GroupId)
+	d.Set("identity_store_id", out.IdentityStoreId)
+	d.Set("description", out.Description)
+	d.Set("display_name", out.DisplayName)
+
+	if err := d.Set("external_ids", flattenExternalIds(out.ExternalIds)); err != nil {
+		return create.DiagError(names.IdentityStore, create.ErrActionSetting, ResNameGroup, d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).IdentityStoreConn
+
+	in := &identitystore.UpdateGroupInput{
+		GroupId:         aws.String(d.Get("group_id").(string)),
+		IdentityStoreId: aws.String(d.Get("identity_store_id").(string)),
+		Operations:      nil,
+	}
+
+	if d.HasChange("display_name") {
+		in.Operations = append(in.Operations, types.AttributeOperation{
+			AttributePath:  aws.String("displayName"),
+			AttributeValue: document.NewLazyDocument(d.Get("display_name").(string)),
+		})
+	}
+
+	if len(in.Operations) > 0 {
+		log.Printf("[DEBUG] Updating IdentityStore Group (%s): %#v", d.Id(), in)
+		_, err := conn.UpdateGroup(ctx, in)
+		if err != nil {
+			return create.DiagError(names.IdentityStore, create.ErrActionUpdating, ResNameGroup, d.Id(), err)
+		}
+	}
+
+	return resourceGroupRead(ctx, d, meta)
+}
+
+func resourceGroupDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).IdentityStoreConn
+
+	log.Printf("[INFO] Deleting IdentityStore Group %s", d.Id())
+
+	_, err := conn.DeleteGroup(ctx, &identitystore.DeleteGroupInput{
+		IdentityStoreId: aws.String(d.Get("identity_store_id").(string)),
+		GroupId:         aws.String(d.Get("group_id").(string)),
+	})
+
+	if err != nil {
+		var nfe *types.ResourceNotFoundException
+		if errors.As(err, &nfe) {
+			return nil
+		}
+
+		return create.DiagError(names.IdentityStore, create.ErrActionDeleting, ResNameGroup, d.Id(), err)
+	}
+
+	return nil
+}
+
+func flattenExternalIds(apiObjects []types.ExternalId) []interface{} {
+	if len(apiObjects) == 0 {
+		return nil
+	}
+
+	var l []interface{}
+
+	for _, apiObject := range apiObjects {
+		if apiObject == (types.ExternalId{}) {
+			continue
+		}
+
+		l = append(l, flattenExternalId(apiObject))
+	}
+
+	return l
+}
+
+func flattenExternalId(apiObject types.ExternalId) map[string]interface{} {
+	if apiObject == (types.ExternalId{}) {
+		return nil
+	}
+
+	m := map[string]interface{}{}
+
+	if v := apiObject.Id; v != nil {
+		m["id"] = aws.ToString(v)
+	}
+
+	if v := apiObject.Issuer; v != nil {
+		m["issuer"] = aws.ToString(v)
+	}
+
+	return m
+}
+
+func resourceGroupParseID(id string) (identityStoreId, groupId string, err error) {
+	parts := strings.Split(id, "/")
+
+	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+		err = errors.New("???")
```

Comment:
This should probably be a message of some kind 😀

Reply:
Nice catch, fixed.